### PR TITLE
Replace initrd img and Add kernel config as optional parameter to transformer 'dom0_binaries'

### PR DESCRIPTION
**Pull request by smit-gardhariya**
Copy kernel config under /boot if passed to the transformer of 'dom0_binaries' type

Copy initrd if passed from runbook else create softlink for existing
initrd image with new kernel version in filename

### CODE REVIEW

File: lisa/transformers/dom0_kernel_installer.py [hunk 1]
```diff
@@ -35,6 +35,14 @@ class BinaryInstallerSchema(BaseInstallerSchema):
         ),
     )
 
+    # kernel config local absolute path
+    kernel_config_path: str = field(
+        default="",
+        metadata=field_metadata(
+            required=True,
```
Comment (pupacha) [inline lisa/transformers/dom0_kernel_installer.py:42]: Does it mean this will be a required field ? If so, we should rather it optional

Comment (smit-gardhariya) [reply lisa/transformers/dom0_kernel_installer.py:42]: @pupacha no, default would be empty string even if we dont pass it from runbook. 

If some value is set, then only we will try to copy the file to target system

Comment (smit-gardhariya) [reply lisa/transformers/dom0_kernel_installer.py:42]: You can refer initrd parameter in this transformer, We are not passing it from runbook in our stack

**Review by pupacha — DISMISSED** [review]

File: lisa/transformers/dom0_kernel_installer.py (hunk 1 above)
Comment (pupacha) [reply lisa/transformers/dom0_kernel_installer.py:42]: Ok. Thanks for confirmation.

**Review by squirrelsc — DISMISSED** [review]

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**